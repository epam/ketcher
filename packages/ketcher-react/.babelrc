{
  "presets": [
    "@babel/env",
    ["@babel/react", { "runtime": "automatic" }],
    "@babel/typescript"
  ],
  "plugins": ["@babel/plugin-transform-runtime"]
}
