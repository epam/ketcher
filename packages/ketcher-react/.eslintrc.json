{
  "root": false,
  "env": {
    "browser": true
  },
  "extends": [
    "plugin:react/jsx-runtime",
    "plugin:jsx-a11y/recommended",
    "plugin:react-hooks/recommended",
    "plugin:jest/recommended"
  ],
  "rules": {
    "jsx-a11y/label-has-associated-control": "warn",
    "jsx-a11y/no-static-element-interactions": "warn",
    "jsx-a11y/no-autofocus": "warn",
    "jsx-a11y/no-noninteractive-element-interactions": "warn",
    "jsx-a11y/alt-text": "warn",
    "jsx-a11y/click-events-have-key-events": "warn"
  },
  "overrides": [
    {
      "files": [
        "**/__tests__/**/*.[jt]s?(x)",
        "**/?(*.)+(spec|test).[jt]s?(x)"
      ],
      "extends": ["plugin:testing-library/react"],
      "rules": {
        "testing-library/no-container": "warn",
        "testing-library/no-node-access": "warn"
      }
    },
    {
      "files": [
        "src/script/builders/ketcher/KetcherBuilder.ts",
        "src/script/editor/Editor.ts",
        "src/script/editor/tool/template.ts",
        "src/script/editor/tool/text.ts",
        "src/script/editor/utils/functionalGroupsTooltip.ts",
        "src/script/ui/App/initApp.tsx",
        "src/script/ui/dialog/template/TemplateTable.tsx",
        "src/script/ui/views/components/ContextMenu/hooks/useBondSGroupAttach.ts",
        "src/script/ui/views/components/ContextMenu/hooks/useBondSGroupEdit.ts",
        "src/script/ui/views/modal/components/toolbox/RgroupLogic/components/IfThenSelect/IfThenSelect.tsx",
        "src/script/ui/views/toolbars/ToolbarGroupItem/ToolbarMultiToolItem/usePortalOpening.ts"
      ],
      "rules": {
        "@typescript-eslint/no-non-null-assertion": "warn"
      }
    },
    {
      "files": [
        "src/script/editor/Editor.ts",
        "src/script/editor/HoverIcon.ts",
        "src/script/editor/tool/Tool.ts",
        "src/script/editor/tool/atom.ts",
        "src/script/editor/tool/bond.ts",
        "src/script/editor/tool/chain.ts",
        "src/script/editor/tool/charge.ts",
        "src/script/editor/tool/eraser.ts",
        "src/script/editor/tool/helper/dropAndMerge.ts",
        "src/script/editor/tool/helper/lasso.ts",
        "src/script/editor/tool/helper/locate.ts",
        "src/script/editor/tool/paste.ts",
        "src/script/editor/tool/reactionarrow.ts",
        "src/script/editor/tool/reactionmap.ts",
        "src/script/editor/tool/reactionplus.ts",
        "src/script/editor/tool/rotate-controller.test.ts",
        "src/script/editor/tool/rotate-controller.ts",
        "src/script/editor/tool/rotate.ts",
        "src/script/editor/tool/select.ts",
        "src/script/editor/tool/sgroup.ts",
        "src/script/editor/tool/simpleobject.ts",
        "src/script/editor/tool/template.ts",
        "src/script/editor/tool/text.ts",
        "src/script/editor/utils/customOnChangeHandler.ts",
        "src/script/ui/App/initApp.tsx",
        "src/script/ui/Portal/Portal.tsx",
        "src/script/ui/action/action.types.ts",
        "src/script/ui/action/isHidden.ts",
        "src/script/ui/component/form/Input/Input.tsx",
        "src/script/ui/component/form/colorPicker/ColorPicker.tsx",
        "src/script/ui/component/structrender.tsx",
        "src/script/ui/component/view/savebutton.tsx",
        "src/script/ui/data/convert/structConverter.ts",
        "src/script/ui/data/schema/options-schema.ts",
        "src/script/ui/dialog/index.ts",
        "src/script/ui/dialog/template/TemplateDialog.tsx",
        "src/script/ui/dialog/template/TemplateTable.tsx",
        "src/script/ui/dialog/toolbox/enhancedStereo/enhancedStereo.tsx",
        "src/script/ui/state/functionalGroups/index.ts",
        "src/script/ui/state/handleHotkeysOverItem.ts",
        "src/script/ui/state/hotkeys.ts",
        "src/script/ui/state/shared.ts",
        "src/script/ui/utils/index.ts",
        "src/script/ui/utils/settingsManager.ts",
        "src/script/ui/views/AppClipArea.tsx",
        "src/script/ui/views/components/Dialog/Dialog.tsx",
        "src/script/ui/views/components/StructEditor/InfoPanel.tsx",
        "src/script/ui/views/components/StructEditor/InfoTooltip.tsx",
        "src/script/ui/views/modal/Modal.tsx",
        "src/script/ui/views/modal/components/Text/Text.tsx",
        "src/script/ui/views/modal/components/document/Open/Open.tsx",
        "src/script/ui/views/modal/components/meta/Settings/Settings.tsx",
        "src/script/ui/views/modal/components/process/Miew/Miew.tsx",
        "src/script/ui/views/modal/components/toolbox/Atom/Atom.container.ts",
        "src/script/ui/views/modal/components/toolbox/Attach/Attach.container.ts",
        "src/script/ui/views/modal/components/toolbox/Automap/Automap.container.ts",
        "src/script/ui/views/modal/components/toolbox/Bond/Bond.container.ts",
        "src/script/ui/views/modal/components/toolbox/FG/RemoveFG.tsx",
        "src/script/ui/views/modal/components/toolbox/RgroupLogic/RgroupLogic.container.ts",
        "src/script/ui/views/modal/components/toolbox/RgroupLogic/components/IfThenSelect/IfThenSelect.tsx",
        "src/script/ui/views/modal/modal.types.ts",
        "src/script/ui/views/toolbars/ArrowScroll/ArrowScroll.tsx",
        "src/script/ui/views/toolbars/BottomToolbar/BottomToolbar.tsx",
        "src/script/ui/views/toolbars/BottomToolbar/TemplatesList/TemplatesList.tsx",
        "src/script/ui/views/toolbars/FloatingTools/FloatingTools.container.ts",
        "src/script/ui/views/toolbars/LeftToolbar/LeftToolbar.container.ts",
        "src/script/ui/views/toolbars/RightToolbar/AtomsList/AtomsList.tsx",
        "src/script/ui/views/toolbars/RightToolbar/RightToolbar.tsx",
        "src/script/ui/views/toolbars/TopToolbar/TopToolbar.container.ts",
        "src/typings.d.ts"
      ],
      "rules": {
        "@typescript-eslint/no-explicit-any": "warn"
      }
    }
  ]
}
