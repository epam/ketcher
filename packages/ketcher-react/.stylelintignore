dist/
node_modules/
style/
