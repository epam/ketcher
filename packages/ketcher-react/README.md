# ketcher-react

[![npm version](https://img.shields.io/npm/v/ketcher-react)](https://www.npmjs.com/package/ketcher-react)
[![Downloads](https://img.shields.io/npm/dm/ketcher-react)](https://www.npmjs.com/package/ketcher-react)
[![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://opensource.org/licenses/Apache-2.0)

Copyright (c) 2021 [EPAM Systems, Inc.](https://www.epam.com/)

Ketcher is an open-source web-based chemical structure editor incorporating high performance, good portability, light weight, and ability to easily integrate into a custom web-application. Ketcher is designed for chemists, laboratory scientists and technicians who draw structures and reactions.

For more details please look at the following [link](https://github.com/epam/ketcher/blob/master/README.md).

The ketcher-react package contains only the functionality necessary to define components. It is used together with [ketcher-core](https://www.npmjs.com/package/ketcher-core) and optionally with [ketcher-standalone](https://www.npmjs.com/package/ketcher-standalone) if standaolone mode is required.

## Installation

The ketcher-react library is available as an [NPM](https://www.npmjs.com/) package. Install it either with NPM:

```sh
npm install --save ketcher-react
```

or [Yarn](https://yarnpkg.com/):

```sh
yarn add ketcher-react
```

## Usage

```js
import { RemoteStructServiceProvider } from 'ketcher-core'

const structServiceProvider = new RemoteStructServiceProvider(
  process.env.REACT_APP_API_PATH!,
  {
    'custom header': 'value' // optionally you can add custom headers object 
  }
)

const MyComponent = () => {
  return (
    <Editor
      staticResourcesUrl={process.env.PUBLIC_URL}
      structServiceProvider={structServiceProvider}
    />
  )
}
```

### Indigo Service

Ketcher uses Indigo Service for server operations.
You may pass it as a property while Editor component is used or just add api_path query parameter:

```sh
<Editor staticResourcesUrl={process.env.PUBLIC_URL} apiPath={link to Indigo service} />
    or
http://localhost:3000/?api_path={link to Indigo service}
```

You can find the instruction for service installation
[here](http://lifescience.opensource.epam.com/indigo/service/index.html).

### 3D Viewer

Ketcher uses Miew-React for viewing and editing data in 3D.
Miew-React package default exports Viewer component which
initializes and renders a Miew instance inside of it

```js
...
import Viewer from 'miew-react'

const MyComponent = () => {
  return <Viewer />
}
...
...
```

You can find the latest version of Miew-React [here](https://github.com/epam/miew/tree/master/packages/miew-react).
The last checked version - [1.0.0](https://www.npmjs.com/package/miew-react).
