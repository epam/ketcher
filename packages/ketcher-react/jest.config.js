module.exports = {
  testMatch: ['**/src/**/?(*.)+(spec|test).[jt]s?(x)'],
  testPathIgnorePatterns: ['fixtures', 'dist', 'node_modules'],
  testEnvironment: 'jsdom',
  transform: {
    '\\.(js|jsx)$': 'babel-jest',
    '^.+\\.(ts|tsx)$': 'ts-jest',
    '\\.svg$': '<rootDir>/testFileTransformer.js'
  },
  moduleNameMapper: {
    '\\.(css|less)$': 'identity-obj-proxy'
  }
}
