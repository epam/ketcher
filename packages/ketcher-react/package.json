{
  "name": "ketcher-react",
  "version": "2.1.0",
  "description": "Web-based molecule sketcher",
  "license": "Apache-2.0",
  "homepage": "http://lifescience.opensource.epam.com/ketcher",
  "author": {
    "name": "EPAM Life Sciences",
    "url": "http://lifescience.opensource.epam.com"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/epam/ketcher.git",
    "directory": "packages/ketcher-react"
  },
  "keywords": [
    "molecule",
    "sketcher",
    "chemistry"
  ],
  "bugs": {
    "url": "https://github.com/epam/ketcher/issues"
  },
  "main": "dist/index.js",
  "module": "dist/index.modern.js",
  "source": "src/index.tsx",
  "engines": {
    "node": ">=10"
  },
  "scripts": {
    "build": "cross-env NODE_ENV=production rollup -c -m true --silent",
    "start": "cross-env NODE_ENV=development rollup -c -m true -w",
    "test": "run-s test:unit test:lint prettier stylelint test:build",
    "test:build": "run-s build",
    "test:lint": "eslint src --ext .ts,.tsx,.js,.jsx",
    "test:unit": "cross-env CI=1 react-scripts test --env=jsdom",
    "test:watch": "react-scripts test --env=jsdom",
    "prettier": "prettier --check \"src/**/*.{js,jsx,json,ts,tsx}\"",
    "prettier:write": "prettier --write \"src/**/*.{js,jsx,json,ts,tsx}\"",
    "stylelint": "stylelint \"src/**/*.{css,less,scss}\" --formatter verbose",
    "stylelint:fix": "stylelint \"src/**/*.{css,less,scss}\" --formatter verbose --fix"
  },
  "peerDependencies": {
    "react": "^16.13.0",
    "react-dom": "^16.13.0"
  },
  "dependencies": {
    "@babel/runtime": "^7.12.5",
    "clsx": "^1.1.1",
    "core-js": "^3.6.5",
    "element-closest-polyfill": "^1.0.2",
    "file-saver": "^2.0.2",
    "font-face-observer": "^1.0.0",
    "hoist-non-react-statics": "^3.3.2",
    "jsonschema": "^1.4.0",
    "ketcher-core": "^1.0.0",
    "lodash": "^4.17.20",
    "normalize.css": "^8.0.1",
    "prop-types": "^15.7.2",
    "raphael": "^2.3.0",
    "react-markdown": "^5.0.2",
    "react-redux": "^7.2.1",
    "react-virtualized": "^9.22.3",
    "redux": "^4.0.5",
    "redux-logger": "^3.0.6",
    "redux-thunk": "^2.3.0",
    "regenerator-runtime": "^0.13.7",
    "remark-gfm": "^1.0.0",
    "remark-parse": "^9.0.0",
    "replace": "^1.2.0",
    "reselect": "^4.0.0",
    "subscription": "^3.0.0",
    "url-search-params-polyfill": "^8.1.0",
    "use-resize-observer": "^7.0.0",
    "w3c-keyname": "^2.2.4",
    "whatwg-fetch": "^3.4.1"
  },
  "devDependencies": {
    "@babel/core": "^7.12.3",
    "@babel/plugin-proposal-class-properties": "^7.12.1",
    "@babel/plugin-proposal-object-rest-spread": "^7.12.1",
    "@babel/plugin-transform-runtime": "^7.12.1",
    "@babel/preset-env": "^7.12.1",
    "@babel/preset-react": "^7.12.5",
    "@babel/preset-typescript": "^7.12.1",
    "@rollup/plugin-babel": "^5.2.1",
    "@rollup/plugin-commonjs": "^16.0.0",
    "@rollup/plugin-json": "^4.1.0",
    "@rollup/plugin-node-resolve": "^10.0.0",
    "@rollup/plugin-replace": "^2.3.4",
    "@rollup/plugin-strip": "^2.0.0",
    "@svgr/rollup": "^5.4.0",
    "@testing-library/jest-dom": "^4.2.4",
    "@testing-library/react": "^9.5.0",
    "@testing-library/user-event": "^7.2.1",
    "@types/jest": "^25.1.4",
    "@types/node": "^12.12.38",
    "@types/react": "^16.9.27",
    "@types/react-dom": "^16.9.7",
    "@typescript-eslint/eslint-plugin": "^4.0.0",
    "@typescript-eslint/parser": "^4.0.0",
    "babel-eslint": "^10.0.0",
    "cross-env": "^7.0.2",
    "eslint": "^6.8.0",
    "eslint-config-react-app": "^6.0.0",
    "eslint-plugin-import": "^2.22.0",
    "eslint-plugin-jest": "^24.0.0",
    "eslint-plugin-jsx-a11y": "^6.3.1",
    "eslint-plugin-promise": "^4.2.1",
    "eslint-plugin-react-hooks": "^4.0.8",
    "eslint-plugin-testing-library": "^3.9.0",
    "less": "^3.12.2",
    "node-sass": "^5.0.0",
    "npm-run-all": "^4.1.5",
    "prettier": "^2.0.4",
    "react-scripts": "^3.4.1",
    "rollup": "^2.33.1",
    "rollup-plugin-cleanup": "^3.2.1",
    "rollup-plugin-copy": "^3.3.0",
    "rollup-plugin-delete": "^2.0.0",
    "rollup-plugin-peer-deps-external": "^2.2.4",
    "rollup-plugin-postcss": "^3.1.8",
    "rollup-plugin-typescript2": "^0.29.0",
    "rollup-plugin-visualizer": "^4.2.0",
    "stylelint": "^13.7.2",
    "stylelint-config-prettier": "^8.0.2",
    "stylelint-config-recommended-scss": "^4.2.0",
    "stylelint-config-standard": "^20.0.0",
    "stylelint-prettier": "^1.1.2",
    "stylelint-scss": "^3.18.0",
    "typescript": "^4.0.5"
  },
  "files": [
    "dist"
  ]
}
