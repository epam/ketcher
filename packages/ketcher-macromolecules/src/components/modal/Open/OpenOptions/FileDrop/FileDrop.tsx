/****************************************************************************
 * Copyright 2021 EPAM Systems
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 ***************************************************************************/
import { useDropzone, DropzoneOptions } from 'react-dropzone';
import { Icon, IconName } from 'ketcher-react';
import React, { useMemo } from 'react';
import styled from '@emotion/styled';
import { OpenOptionText, DisabledText } from '../OpenOptions';

export type FileDropProps = {
  buttonLabel: string;
  textLabel: string;
  iconName: IconName;
  disabled?: boolean;
  disabledText?: string;
} & DropzoneOptions;

const baseStyle = {
  width: '100%',
  height: '100%',
  display: 'flex',
  alignItems: 'center',
  flexDirection: 'column',
  justifyContent: 'space-between',
};

interface StyledIconProps {
  disabled?: boolean;
}

const StyledIcon = styled(Icon)<StyledIconProps>`
  filter: ${({ disabled }) => (disabled ? 'grayscale(1)' : '')};
  opacity: ${({ disabled }) => (disabled ? '0.6' : '1')};
`;

const activeStyle = {
  backgroundColor: '#F8FEFFFF',
};

const ButtonContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;

  & > span {
    font-size: ${({ theme }) => theme.ketcher.font.size.small};
    color: ${({ theme }) => theme.ketcher.color.text.primary};
    opacity: 50%;
  }
  & > svg {
    margin-bottom: 8px;
  }
`;

const FileDrop = ({
  textLabel,
  iconName,
  disabled,
  disabledText,
  ...rest
}: FileDropProps) => {
  const { getRootProps, getInputProps, isDragActive, open } = useDropzone({
    multiple: false,
    noClick: true,
    disabled,
    ...rest,
  });

  const style = useMemo(
    () => ({
      ...baseStyle,
      ...(isDragActive ? activeStyle : {}),
    }),
    [isDragActive],
  ) as React.CSSProperties;

  return (
    <div {...getRootProps({ style })} onClick={open}>
      <input {...getInputProps()} />
      <StyledIcon name={iconName} disabled={disabled} />
      {disabled ? (
        <DisabledText>{disabledText}</DisabledText>
      ) : (
        <>
          <ButtonContainer>
            {textLabel && <span>{textLabel}</span>}
          </ButtonContainer>
          <OpenOptionText>Open from file</OpenOptionText>
        </>
      )}
    </div>
  );
};

export { FileDrop };
