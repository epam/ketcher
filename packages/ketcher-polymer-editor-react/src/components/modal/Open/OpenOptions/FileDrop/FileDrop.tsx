/****************************************************************************
 * Copyright 2021 EPAM Systems
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 ***************************************************************************/
import { useDropzone, DropzoneOptions } from 'react-dropzone';
import { Icon, IconName } from 'ketcher-react';
import React, { useMemo } from 'react';
import { ActionButton } from 'components/shared/actionButton';
import styled from '@emotion/styled';

export type FileDropProps = {
  buttonLabel: string;
  textLabel: string;
  iconName: IconName;
  disabled?: boolean;
  disabledText?: string;
} & DropzoneOptions;

const baseStyle = {
  width: '100%',
  height: '100%',
  display: 'flex',
  alignItems: 'center',
  flexDirection: 'column',
  justifyContent: 'space-around',
};

const activeStyle = {
  backgroundColor: '#F8FEFFFF',
};

const ButtonContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: 8px;

  & > span {
    padding-top: 5px;
    font-size: ${({ theme }) => theme.ketcher.font.size.small};
  }
  & > svg {
    margin-bottom: 8px;
  }
`;

const FileDrop = ({
  buttonLabel,
  textLabel,
  iconName,
  disabled,
  disabledText,
  ...rest
}: FileDropProps) => {
  const { getRootProps, getInputProps, isDragActive, open } = useDropzone({
    multiple: false,
    noClick: true,
    disabled,
    ...rest,
  });

  const style = useMemo(
    () => ({
      ...baseStyle,
      ...(isDragActive ? activeStyle : {}),
    }),
    [isDragActive],
  ) as React.CSSProperties;

  return (
    <div {...getRootProps({ style })}>
      <input {...getInputProps()} />
      <ButtonContainer>
        <ActionButton
          label={buttonLabel}
          clickHandler={open}
          disabled={disabled}
        />
        {textLabel && <span>{textLabel}</span>}
      </ButtonContainer>
      {disabled ? <p>{disabledText}</p> : <Icon name={iconName} />}
    </div>
  );
};

export { FileDrop };
