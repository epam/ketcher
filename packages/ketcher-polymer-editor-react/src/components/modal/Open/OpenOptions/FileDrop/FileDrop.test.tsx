/****************************************************************************
 * Copyright 2021 EPAM Systems
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 ***************************************************************************/
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { IconName } from 'ketcher-react';

import { FileDrop, FileDropProps } from './FileDrop';

const mockProps = {
  buttonLabel: 'Open from file',
  textLabel: 'or drag file here',
  iconName: 'arrow-upward' as IconName,
  onDropAccepted: jest.fn(),
} as FileDropProps;

const mockOptionalProps = {
  disabled: true,
  disabledText: 'Mock disabled text',
};

const mockFile = new File(['ketcher mol file'], 'ketcher.mol', {
  type: 'chem/mol',
});

describe('FileDrop component', () => {
  it('should render correctly', () => {
    expect(
      render(withThemeProvider(<FileDrop {...mockProps} />)),
    ).toMatchSnapshot();
  });

  it('should render correctly with required props', () => {
    render(withThemeProvider(<FileDrop {...mockProps} />));

    expect(
      screen.getByRole('button', { name: mockProps.buttonLabel }),
    ).toBeInTheDocument();
    expect(screen.getByText(mockProps.textLabel)).toBeInTheDocument();
  });

  it('should render correctly with optional props', () => {
    render(
      withThemeProvider(<FileDrop {...mockProps} {...mockOptionalProps} />),
    );

    expect(
      screen.getByRole('button', { name: mockProps.buttonLabel }),
    ).toBeInTheDocument();
    expect(screen.getByText(mockProps.textLabel)).toBeInTheDocument();
    expect(
      screen.getByText(mockOptionalProps.disabledText),
    ).toBeInTheDocument();
  });

  it('should upload file', async () => {
    const { container } = render(
      withThemeProvider(<FileDrop {...mockProps} />),
    );
    const input = container.querySelector(
      'input[type=file]',
    ) as HTMLInputElement;

    userEvent.upload(input, mockFile);

    await waitFor(() => {
      expect(input.files && input.files[0]).toBe(mockFile);
    });
    await waitFor(() => {
      expect(input.files && input.files.item(0)).toBe(mockFile);
    });
    await waitFor(() => {
      expect(input.files && input.files).toHaveLength(1);
    });
  });

  it('accepted file callback should be called after file is uploaded', async () => {
    const { container } = render(
      withThemeProvider(<FileDrop {...mockProps} />),
    );
    const input = container.querySelector(
      'input[type=file]',
    ) as HTMLInputElement;

    userEvent.upload(input, mockFile);

    await waitFor(() => {
      expect(mockProps.onDropAccepted).toBeCalled();
    });
  });
});
