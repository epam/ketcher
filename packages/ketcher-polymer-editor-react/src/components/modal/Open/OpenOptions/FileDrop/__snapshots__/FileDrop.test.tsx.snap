// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileDrop component should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        role="button"
        style="width: 100%; height: 100%; display: flex; align-items: center; flex-direction: column; justify-content: space-around;"
        tabindex="0"
      >
        <input
          autocomplete="off"
          style="display: none;"
          tabindex="-1"
          type="file"
        />
        <div
          class="css-1gnig72"
        >
          <button
            class="MuiButtonBase-root css-1u2dyik-MuiButtonBase-root"
            role="button"
            tabindex="0"
            title="Open from file"
            type="button"
          >
            Open from file
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <span>
            or drag file here
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      role="button"
      style="width: 100%; height: 100%; display: flex; align-items: center; flex-direction: column; justify-content: space-around;"
      tabindex="0"
    >
      <input
        autocomplete="off"
        style="display: none;"
        tabindex="-1"
        type="file"
      />
      <div
        class="css-1gnig72"
      >
        <button
          class="MuiButtonBase-root css-1u2dyik-MuiButtonBase-root"
          role="button"
          tabindex="0"
          title="Open from file"
          type="button"
        >
          Open from file
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <span>
          or drag file here
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
